src/cli.js

```diff
@@ -611,15 +611,21 @@ var exports = {
     }
 
     var filename;
+    var lintStdinFile;
 
     // There is an if(filename) check in the lint() function called below.
     // passing a filename of undefined is the same as calling the function
     // without a filename.  If there is no opts.filename, filename remains
     // undefined and lint() is effectively called with 4 parameters.
     if (opts.filename) {
-      filename = path.resolve(opts.filename);
+      filename = path.relative(path.resolve('.'), path.resolve(opts.filename));
+      var ignores = !opts.ignores ? loadIgnores({ cwd: opts.cwd }) :
+                                  opts.ignores.map(function(target) {
+                                    return path.resolve(target);
```

Comment:
Since the strings in `opts.ignores` are patterns and not literal paths, I'm a little reluctant to perform path operations on them.

Reply:
I actually copy/pasted this code from the `gather` function (line 574 in my checkout), which is probably bad practice anyway - what would the correct thing to be doing here be?

Reply:
Ah, thanks for pointing that out. I think consistency has to win out over my more theoretical concerns, so we'll stick with the existing pattern.